src/hotspot/share/oops/methodData.hpp

```diff
@@ -1983,8 +1984,8 @@ class MethodData : public Metadata {
     } _trap_hist;
 
   public:
-    MDHeader() : _nof_decompiles(0), _nof_overflow_recompiles(0), _nof_overflow_traps(0) {
-      assert(sizeof(_trap_hist) % sizeof(HeapWord) == 0, "align");
+    CompilerCounters() : _nof_decompiles(0), _nof_overflow_recompiles(0), _nof_overflow_traps(0) {
```

Comment:
This doesn't initialize _creation_mileage.  Is that a bug?  Should that be a constructor argument?

Reply:
> This doesn't initialize _creation_mileage. Is that a bug? Should that be a constructor argument?

I don't think there's a bug since `MethodData` initializes it right away (and `ciMethodData` bulk copies it from the original version), but agree passing it as a constructor argument is cleaner. Updated version posted.